### PR TITLE
Generate: better warnings with pipelines

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Addresses comments in https://github.com/huggingface/transformers/issues/23054

This PR adds the following enhancements to generate-related pipeline warnings:
1. Clarifies the `max_length` reduction suggestion in the summarization pipeline
2. Also pipes task-specific configuration to `generation_config` (when applicable), which fixes the warning about relying on `model.config` to parameterize `generate`

### CODE REVIEW

**Review by Narsil — APPROVED** [review]
LGTM, Thank you for this !

File: src/transformers/pipelines/text2text_generation.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def check_inputs(self, input_length: int, min_length: int, max_length: int) -> b
 
         if input_length < max_length:
             logger.warning(
-                f"Your max_length is set to {max_length}, but your input_length is only {input_length}. You might "
+                f"Your max_length is set to {max_length}, but your input_length is only {input_length}. Since this is a summarization task, where outputs shorter than the input are typically wanted, you might "
```
Comment (sgugger) [inline src/transformers/pipelines/text2text_generation.py:276]: Can you just respect the 119 char length here (not sure why black didn't do it).

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

File: src/transformers/pipelines/text2text_generation.py (hunk 1 above)
Comment (gante) [reply src/transformers/pipelines/text2text_generation.py:276]: Interesting, I relied on `black` to fix it for me :D 

pushing a fix

**Pull request merged**